enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,727 @@
+
+### Drawbacks
+
+Not relevant here.
+
+## Open Questions [optional]
+
+### Autoscaling Future
+
+Autoscaling will be a followup enhancement. 
+Kueue does not provide much guidance on the safe enablement of autoscaling.
+We know that we need a secure way of enabling autoscaler and we should think through that in more detail.
+
+### RHOAI and Kueue Integration
+
+RHOAI is using Kueue as a GA product. We still need to figure out the path with RHOAI and OCP Kueue.
+
+### Expermental Feature Support
+
+Kueue has a concept of feature gates in their configuration API. These are a series of advanced features.
+The development of Kueue is quite fast and many of these features are not yet GA. 
+We are engaging with upstream to avoid permanent betas and to focus on graduating feature gates.
+
+Meanwhile, there are cases where one would want to test alpha features or beta features. 
+To do this, we want to provide an expermental stream in OLM that will allow one to change feature gates and set non standard options.
+We will achieve this by building a special alpha bundle that sets a flag in our deployment that will allow the changing of advanced functionality.
+
+As of now there are a few other options we need to explore.
+
+- Dedicated olm bundle for expermental features
+- Feature gates in operator to allow one to change functionality (same bundle)
+- FeatureSet at OCP level
+
+## Test Plan
+
+There are three areas we want to increase testing.
+
+Upstream, downstream operand and the operator.
+
+### Upstream
+
+The Kueue Operator will depend on the Cert Manager integration. 
+Kueue does not test or use heavily cert manager so we will work with upstream to add tests to
+confirm cert manager functionality.
+
+All features/bugs should be implemented and tested in Kueue upstream.
+
+Kueue runs their e2e tests on all supported versions of Kubernetes (n-3) with its two latest releases.
+
+#### Downstream Operand
+
+Kueue operand will need to be tested on all supported versions of Openshift that the operator will support.
+We have to change the operand to be compliant with Red Hat policies so we should run the upstream tests with the operand images.
+
+#### Operator
+
+There will be e2e tests testing the installion of the operator based on user provided configurations.
+The operator will also test the installation of Kueue and the removing of Kueue based on OLM bundles.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- Konflux releases of operator
+- E2E Testing
+- Operator functionality complete
+- Documentation
+- Telemetry
+
+### RHOAI GA Adoption
+
+- RHOAI is able to switch their Kueue deployment for Openshift Kueue
+- Feature parity with their existing solution
+- We are discussing them adopting this as GA before we release the operator as GA for everyone.
+
+### Tech Preview -> GA
+
+- Once kueue APIs are V1 we can GA the kueue operator.
```

Comment:
Do we have a timeline for this?

Reply:
As with any upstream project, that is dictated by the maintainers.

We know that there may be a new api bump from v1beta1 -> v1beta2 in 2025 timeframe. I think maybe 2026 there are discussions to promote Kueue to v1.
But there are also a lot of existing features in Kueue so there was push back to graduate the APIs to v1.

Reply:
Is maintaining this as TP for an extended period going to cause more or less pain for us then GA'ing on the beta APIs and taking the hit of having to lifecycle and support the APIs as they evolve?

Reply:
I pushed some options. There was a discussion yesterday where we GA the GAish apis like ClusterQueues, LocalQueues, Workloads etc.

cc @mrunalp